tablecodec/tablecodec.go

```diff
@@ -138,6 +138,10 @@ func DecodeRecordKey(key kv.Key) (tableID int64, handle kv.Handle, err error) {
 		return 0, nil, errors.Trace(err)
 	}
 
+	if len(key) == prefixLen {
```

Comment:
I think key here has been changed by line 135 and line 136. Maybe use `k` instead of `key`?